gadget/gadget_test.go

```diff
@@ -2400,3 +2401,120 @@ volumes:
 	err = gadget.IsCompatible(gi, giNew)
 	c.Check(err, IsNil)
 }
+
+func (s *gadgetYamlTestSuite) TestKernelCommandLineBasic(c *C) {
+	for _, tc := range []struct {
+		files [][]string
+
+		cmdline string
+		full    bool
+		err     string
+	}{{
+		files: [][]string{
+			{"cmdline.extra", "   foo bar baz just-extra\n"},
+		},
+		cmdline: "foo bar baz just-extra", full: false,
+	}, {
+		files: [][]string{
+			{"cmdline.full", "    foo bar baz full\n"},
+		},
+		cmdline: "foo bar baz full", full: true,
+	}, {
+		// no cmdline
+		files: nil,
+		err:   gadget.ErrNoKernelCommandline.Error(),
```

Comment:
I think it's better to have the expected string here in case it changes, otherwise we are just comparing that a variable is equal to the same variable imported here and would miss the fact that it changed

Reply:
Added